// @scalar/api-reference/dist/components/Sidebar/SidebarElement.vue2.js
import { defineComponent as A, openBlock as n, createElementBlock as l, createElementVNode as a, normalizeClass as T, unref as m, withModifiers as V, createVNode as b, withCtx as E, createTextVNode as c, toDisplayString as h, createCommentVNode as s, createBlock as N, renderSlot as v } from "vue";
import { ScalarSidebarGroupToggle as O, ScalarIcon as U } from "@scalar/components";
import { combineUrlAndPath as H } from "@scalar/oas-utils/helpers";
import { useConfig as I } from "../../hooks/useConfig.js";
import P from "./SidebarHttpBadge.vue.js";
import { sleep as C } from "../../helpers/sleep.js";
import { useNavState as D } from "../../hooks/useNavState.js";
import { scrollToId as L } from "../../helpers/scrollToId.js";
const M = ["id"], R = {
  key: 0,
  class: "sidebar-heading-chevron"
}, z = ["aria-expanded"], F = ["href", "tabindex"], G = { class: "sidebar-heading-link-title" }, j = {
  key: 1,
  class: "sidebar-heading-link-method"
}, q = {
  key: 1,
  class: "action-menu"
}, x = /* @__PURE__ */ A({
  __name: "SidebarElement",
  props: {
    id: {},
    item: {},
    isActive: { type: Boolean },
    hasChildren: { type: Boolean },
    open: { type: Boolean }
  },
  emits: ["toggleOpen"],
  setup(k, { emit: y }) {
    const i = k, d = y, { getFullHash: u, isIntersectionEnabled: r, pathRouting: p, replaceUrlState: w } = D(), S = I(), f = async () => {
      var e, t;
      i.hasChildren && d("toggleOpen"), (t = (e = i.item) == null ? void 0 : e.select) == null || t.call(e), i.open && (r.value = !1, await C(100), r.value = !0);
    }, B = () => {
      if (p.value)
        return H(p.value.basePath, i.item.id);
      if (typeof window < "u") {
        const e = new URL(window.location.href);
        return e.hash = u(i.item.id), `${e.pathname}${e.search}${e.hash}`;
      }
      return `#${u(i.item.id)}`;
    }, $ = async (e) => {
      var t, o;
      p.value && (e.preventDefault(), i.hasChildren && d("toggleOpen"), (o = (t = i.item) == null ? void 0 : t.select) == null || o.call(t), d("toggleOpen"), r.value = !1, w(i.item.id), L(i.item.id), await C(100), r.value = !0);
    };
    return (e, t) => {
      var o, g;
      return n(), l("li", {
        id: e.id,
        class: "sidebar-group-item"
      }, [
        a("div", {
          class: T(["sidebar-heading", {
            "sidebar-group-item__folder": e.hasChildren,
            active_page: e.isActive,
            deprecated: e.item.deprecated ?? !1
          }]),
          onClick: f
        }, [
          e.hasChildren && !m(S).defaultOpenAllTags ? (n(), l("p", R, [
            a("button", {
              "aria-expanded": e.open,
              class: "toggle-nested-icon",
              type: "button",
              onClick: V(f, ["stop"])
            }, [
              b(m(O), { open: e.open }, {
                label: E(() => [
                  c(h(e.open ? "Collapse" : "Expand") + " " + h(e.item.title), 1)
                ]),
                _: 1
              }, 8, ["open"])
            ], 8, z),
            t[0] || (t[0] = c("   "))
          ])) : s("", !0),
          a("a", {
            class: "sidebar-heading-link",
            href: B(),
            tabindex: e.hasChildren ? -1 : 0,
            onClick: $
          }, [
            (g = (o = e.item) == null ? void 0 : o.icon) != null && g.src ? (n(), N(m(U), {
              key: 0,
              class: "sidebar-icon",
              icon: e.item.icon.src
            }, null, 8, ["icon"])) : s("", !0),
            a("p", G, h(e.item.title), 1),
            e.item.httpVerb && !e.hasChildren ? (n(), l("p", j, [
              t[1] || (t[1] = c("   ")),
              t[2] || (t[2] = a("span", { class: "sr-only" }, "HTTP Method: ", -1)),
              b(P, {
                active: e.isActive,
                method: e.item.httpVerb
              }, null, 8, ["active", "method"])
            ])) : s("", !0)
          ], 8, F)
        ], 2),
        e.open ? v(e.$slots, "default", { key: 0 }, void 0, !0) : s("", !0),
        e.$slots["action-menu"] ? (n(), l("div", q, [
          v(e.$slots, "action-menu", {}, void 0, !0)
        ])) : s("", !0)
      ], 8, M);
    };
  }
});
export {
  x as default
};
