// @scalar/api-reference/dist/features/ExampleRequest/ExampleRequest.vue2.js
import { defineComponent as z, useId as A, computed as c, ref as G, watch as L, unref as l, openBlock as d, createBlock as x, withCtx as r, createVNode as a, createElementBlock as k, Fragment as N, createTextVNode as f, toDisplayString as b, createElementVNode as h, renderSlot as K, createCommentVNode as C } from "vue";
import { useWorkspace as Q } from "@scalar/api-client/store";
import { getSnippet as X } from "@scalar/api-client/views/Components/CodeSnippet";
import { filterSecurityRequirements as Y } from "@scalar/api-client/views/Request/RequestSection";
import { ScalarCodeBlock as Z } from "@scalar/components";
import E from "../../components/ScreenReader.vue.js";
import { useConfig as _ } from "../../hooks/useConfig.js";
import ee from "./ExamplePicker.vue.js";
import te from "./TextSelect.vue.js";
/* empty css                */
import B from "../../components/Card/Card.vue.js";
import oe from "../../components/Card/CardHeader.vue.js";
import T from "../../components/HttpMethod/HttpMethod.vue.js";
import V from "../../components/Card/CardContent.vue.js";
import le from "../../components/Card/CardFooter.vue.js";
import { useExampleStore as re } from "../../legacy/stores/useExampleStore.js";
import { useHttpClientStore as ae } from "../../stores/useHttpClientStore.js";
const ne = ["id"], se = ["id"], ie = {
  key: 0,
  class: "request-card-footer-addon"
}, ue = { class: "request-header" }, $e = /* @__PURE__ */ z({
  __name: "ExampleRequest",
  props: {
    operation: {},
    server: {},
    collection: {},
    fallback: { type: Boolean },
    transformedOperation: {}
  },
  setup(n) {
    const { selectedExampleKey: j, operationId: H } = re(), { requestExamples: R, securitySchemes: S } = Q(), w = _(), {
      httpClient: i,
      setHttpClient: J,
      availableTargets: q,
      httpTargetTitle: I,
      httpClientTitle: U
    } = ae(), y = A(), m = c(() => {
      var t;
      const e = ["x-custom-examples", "x-codeSamples", "x-code-samples"];
      for (const o of e)
        if ((t = n.transformedOperation.information) != null && t[o])
          return [...n.transformedOperation.information[o]];
      return [];
    }), s = G(
      // Default to first custom example
      m.value.length ? {
        targetKey: "customExamples",
        clientKey: 0
      } : (
        // Otherwise use the globally selected HTTP client
        {
          targetKey: i.targetKey,
          clientKey: i.clientKey
        }
      )
    );
    L(i, () => {
      s.value = {
        targetKey: i.targetKey,
        clientKey: i.clientKey
      };
    });
    const $ = c(
      () => {
        var e, t, o, u;
        return Object.keys(
          ((u = (o = (t = (e = n.transformedOperation.information) == null ? void 0 : e.requestBody) == null ? void 0 : t.content) == null ? void 0 : o["application/json"]) == null ? void 0 : u.examples) ?? {}
        ).length > 1;
      }
    ), F = () => {
      var g;
      if (s.value.targetKey === "customExamples")
        return ((g = m.value[s.value.clientKey]) == null ? void 0 : g.source) ?? "";
      const e = i.clientKey, t = i.targetKey, o = R[n.operation.examples[0]];
      if (!o)
        return "";
      const u = Y(
        n.operation.security || n.collection.security,
        n.collection.selectedSecuritySchemeUids,
        S
      ), [p, v] = X(t, e, {
        operation: n.operation,
        example: o,
        server: n.server,
        securitySchemes: u
      });
      return p ? p.message ?? "" : v;
    }, O = c(() => {
      try {
        return F();
      } catch (e) {
        return console.error("[generateSnippet]", e), "";
      }
    }), M = c(() => {
      var t, o;
      const e = (
        // Specified language
        ((t = s.value) == null ? void 0 : t.targetKey) === "customExamples" ? ((o = m.value[s.value.clientKey]) == null ? void 0 : o.lang) ?? "plaintext" : (
          // Or language for the globally selected HTTP client
          i.targetKey
        )
      );
      return e === "shell" && O.value.includes("curl") ? "curl" : e === "Objective-C" ? "objc" : e;
    }), P = c(
      () => Object.values(S).flatMap((e) => e.type === "apiKey" ? e.value : (e == null ? void 0 : e.type) === "http" ? [
        e.token,
        e.password,
        btoa(`${e.username}:${e.password}`)
      ] : e.type === "oauth2" ? Object.values(e.flows).map((t) => t.token) : [])
    ), D = c(() => {
      const e = q.value.map((t) => ({
        value: t.key,
        label: t.title,
        options: t.clients.map((o) => ({
          value: JSON.stringify({
            targetKey: t.key,
            clientKey: o.client
          }),
          label: o.title
        }))
      }));
      return m.value.length && e.unshift({
        value: "customExamples",
        label: "Examples",
        options: m.value.map((t, o) => ({
          value: JSON.stringify({
            targetKey: "customExamples",
            clientKey: o
          }),
          label: t.label ?? t.lang ?? `Example #${o + 1}`
        }))
      }), e;
    });
    function W(e) {
      const t = JSON.parse(e);
      t.targetKey === "customExamples" ? s.value = t : J(t);
    }
    return (e, t) => l(q).length || m.value.length ? (d(), x(l(B), {
      key: 0,
      "aria-labelledby": `${l(y)}-header`,
      class: "dark-mode",
      role: "region"
    }, {
      default: r(() => [
        a(l(oe), { muted: "" }, {
          actions: r(() => [
            a(te, {
              class: "request-client-picker",
              controls: `${l(y)}-example`,
              modelValue: JSON.stringify(s.value),
              options: D.value,
              "onUpdate:modelValue": W
            }, {
              default: r(() => [
                s.value.targetKey === "customExamples" ? (d(), k(N, { key: 0 }, [
                  a(E, null, {
                    default: r(() => t[2] || (t[2] = [
                      f("Selected Example:")
                    ])),
                    _: 1
                  }),
                  f(" " + b(m.value[s.value.clientKey].label ?? "Example"), 1)
                ], 64)) : (d(), k(N, { key: 1 }, [
                  a(E, null, {
                    default: r(() => t[3] || (t[3] = [
                      f("Selected HTTP client:")
                    ])),
                    _: 1
                  }),
                  f(" " + b(l(I)) + " " + b(l(U)), 1)
                ], 64))
              ]),
              _: 1
            }, 8, ["controls", "modelValue", "options"])
          ]),
          default: r(() => [
            h("div", {
              id: `${l(y)}-header`,
              class: "request-header"
            }, [
              a(E, null, {
                default: r(() => t[1] || (t[1] = [
                  f("Request Example for")
                ])),
                _: 1
              }),
              a(l(T), {
                as: "span",
                class: "request-method",
                method: e.operation.method
              }, null, 8, ["method"]),
              K(e.$slots, "header", {}, void 0, !0)
            ], 8, ne)
          ]),
          _: 3
        }),
        a(l(V), {
          borderless: "",
          class: "request-editor-section custom-scroll",
          frameless: ""
        }, {
          default: r(() => [
            h("div", {
              id: `${l(y)}-example`,
              class: "code-snippet"
            }, [
              a(l(Z), {
                class: "bg-b-2 -outline-offset-2",
                content: O.value,
                hideCredentials: P.value,
                lang: M.value,
                lineNumbers: ""
              }, null, 8, ["content", "hideCredentials", "lang"])
            ], 8, se)
          ]),
          _: 1
        }),
        ($.value || !l(w).hideTestRequestButton) && e.$slots.footer ? (d(), x(l(le), {
          key: 0,
          class: "request-card-footer",
          contrast: ""
        }, {
          default: r(() => {
            var o, u, p, v;
            return [
              $.value ? (d(), k("div", ie, [
                a(ee, {
                  class: "request-example-selector",
                  examples: ((v = (p = (u = (o = e.transformedOperation.information) == null ? void 0 : o.requestBody) == null ? void 0 : u.content) == null ? void 0 : p["application/json"]) == null ? void 0 : v.examples) ?? [],
                  "onUpdate:modelValue": t[0] || (t[0] = (g) => (j.value = g, H.value = e.operation.operationId))
                }, null, 8, ["examples"])
              ])) : C("", !0),
              K(e.$slots, "footer", {}, void 0, !0)
            ];
          }),
          _: 3
        })) : C("", !0)
      ]),
      _: 3
    }, 8, ["aria-labelledby"])) : e.fallback ? (d(), x(l(B), {
      key: 1,
      class: "dark-mode"
    }, {
      default: r(() => [
        a(l(V), { class: "request-card-simple" }, {
          default: r(() => [
            h("div", ue, [
              a(l(T), {
                as: "span",
                class: "request-method",
                method: e.operation.method
              }, null, 8, ["method"]),
              K(e.$slots, "header", {}, void 0, !0)
            ]),
            K(e.$slots, "footer", {}, void 0, !0)
          ]),
          _: 3
        })
      ]),
      _: 3
    })) : C("", !0);
  }
});
export {
  $e as default
};
